### PR TITLE
Show a notice when user taps to add a disabled block option

### LINKED ISSUE #3055: Show notice when user taps a disabled block option
**Describe the solution you'd like**
Add either a Snackbar or Compact notice when the user tries to tap a disabled block option as [requested in this PR](https://github.com/WordPress/gutenberg/pull/28339). This would give the user some feedback as to why this option is disabled. The message could be something along the lines of  "Only one Read More block is allowed on a [post/page]".

### PR DESCRIPTION
**Pull request by dnalves**
* Gutenberg PR: https://github.com/WordPress/gutenberg/pull/35021
* WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15435
* WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17293

Fixes #3055

### Description
Whenever a block doesn't support multiple instances and is already added to
the post, its icon is greyed out and it can't be added. With this commit,
when the user taps on it, it shows a Notice with the information on why
that block can't be added.

### To test
1. Add a "More" block to the Post
2. Tap the "+" button to add a block
3. See that the "More" button is greyed out
4. Tap on the "More" button and see that a Notice is displayed on top 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "gutenberg"]
 	path = gutenberg
-	url = ../../WordPress/gutenberg.git
```
Comment (jd-alexander) [inline .gitmodules]: @dnalves we don't want to merge this submodule URL change into `develop` 👍🏾 We can remove this and update the Gutenberg ref to the merge commit once the changes in the Gutenberg PR are merged in.

Comment (dnalves) [reply .gitmodules]: My bad, I didn't intend to push this change, it was mainly for dev purposes. Thanks for spotting it!
Fixed it here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4005/commits/52be2cd79cc744e8c4453920f5ad1209d977f84b

**Comment by fluiddot** [comment]
@dnalves following this comment https://github.com/wordpress-mobile/WordPress-iOS/pull/17293#issuecomment-944334508, I'd recommend running the command `npm run bundle` to generate the JS bundle which is required when we introduce changes in the main apps, specifically in WPiOS.

Heads up that you'll probably notice modifications in the strings localization files (`bundle/android/strings.xml` and `bundle/ios/GutenbergNativeTranslations.swift`), due to a known issue, the bundle generation command is removing some strings that are required, so we need to verify the actual string modifications and only commit the required changes in those files. In case it helps, you can see in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4115/commits/998980982afa971a2de653f0c23bac520f0e43c8) how I solved it when I generated the JS bundle for testing these changes.

**Comment by dnalves** [comment]
> @dnalves following this comment [wordpress-mobile/WordPress-iOS#17293 (comment)](https://github.com/wordpress-mobile/WordPress-iOS/pull/17293#issuecomment-944334508), I'd recommend running the command `npm run bundle` to generate the JS bundle which is required when we introduce changes in the main apps, specifically in WPiOS.
> 
> Heads up that you'll probably notice modifications in the strings localization files (`bundle/android/strings.xml` and `bundle/ios/GutenbergNativeTranslations.swift`), due to a known issue, the bundle generation command is removing some strings that are required, so we need to verify the actual string modifications and only commit the required changes in those files. In case it helps, you can see in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4115/commits/998980982afa971a2de653f0c23bac520f0e43c8) how I solved it when I generated the JS bundle for testing these changes.

Thank you so much @fluiddot ! You helped me immensely today! ☺️ 
I just pushed the change with the bundle, your instructions were on point! :D

**Comment by dcalhoun** [comment]
Hey @jd-alexander and @dnalves. We will cut the 1.65.0 release on 28 Oct. I plan to circle back and bump this PR to the next milestone then, but please let me know if you’d rather us work to include this PR in 1.65.0. Thanks!

**Comment by fluiddot** [comment]
Hey @jd-alexander and @dnalves  !
We will cut the 1.66.0 release on 2021-11-11. I plan to circle back and bump this PR to the next milestone then, but please let me know if you’d rather us work to include this PR in 1.66.0. Thanks!

**Comment by SiobhyB** [comment]
👋 @dnalves, we plan to cut the 1.67.0 release around 10:00 UTC tomorrow, on Thursday 25th November. I plan to circle back and bump this PR to the next milestone then, but please let me know if you'd rather us work to include this PR in 1.67.0. 

As a reminder, changes in the next (1.68.0) release won't be shipped to users until the 10th January.

Thanks!

**Pull request closed without merging**